setup.py

```diff
@@ -31,18 +31,20 @@
 
 def get_all_extras() -> Dict:
     cli_deps = [
-        "click==8.0.2",
-        "pyyaml==6.0.1",
+        "click>=8.0.2,<9",
+        "pyyaml>=6.0.1,<9",
         "packaging>=23.1,<24.0",
-        "pytest>=7.0.0,<7.3.0",
         "coverage>=6.4.4,<8.0.0",
         "jsonschema<4.4.0,>=4.3.0",
         "semver>=2.9.1,<3.0.0",
     ]
+    test_deps = [
+        "pytest>=7.0.0,<8",
+    ]
```

Comment:
Pytest is a framework dependency, it's used in the `aea test` module so making this change will lead to things breaking downstream. Although, this change can be introduced in a major release so I would suggest  you open an issue for this.

Reply:
Reverted as well